app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditViewModel.java

```diff
@@ -72,43 +58,34 @@ public void handleAddBook(
-            @Nullable Uri newUri, OnSuccessCallbackWithMessage<Book> successCallback, OnFailCallback failCallback
+            @Nullable Uri newUri, OnSuccessCallbackWithMessage<Book> successCallback, OnFailCallbackWithMessage<BOOK_ERROR> failCallback
     ) {
-            String bookId = oldBook.getId();
-            bookRepository.editBook(bookId, isbn, title, author, description).addOnCompleteListener(
-                    editBookTask -> {
-                        if (editBookTask.isSuccessful()) {
-                            Book book = new Book(bookId, isbn, title, author, description, oldBook.getOwner(), BOOK_STATUS.AVAILABLE, newUri);
-                            Uri oldUri = oldBook.getImageUri();
-                            if (newUri != null) {
-                                // image not changed
-                                if (oldUri != null && oldUri.equals(newUri)) {
-                                    successCallback.run(book);
-                                } else { // image is being added or replaced
-                                    String imageName = bookId + "cover";
-                                    // addImage will also replace if file with imageName already exist
-                                    bookRepository.addImage(imageName, newUri).addOnCompleteListener(
-                                            addImageTask -> {
-                                                if (addImageTask.isSuccessful()) {
-                                                    bookRepository.addImageNameToBook(bookId, imageName).addOnCompleteListener(
-                                                            addNameTask -> successCallback.run(book)
-                                                    );
-                                                } else {
-                                                    failCallback.run(BOOK_ERROR.FAIL_TO_ADD_IMAGE);
-                                                }
-                                            }
-                                    );
-                                }
-                            } else if (oldUri != null && newUri == null){
-                                // user deletes image (not yet implemented)
-                            } else { // no image before edit and no image after edit
-                                successCallback.run(book);
-                            }
+        String bookId = oldBook.getId();
+        bookRepository.editBook(bookId, isbn, title, author, description).addOnCompleteListener(
+                editBookTask -> {
+                    if (editBookTask.isSuccessful()) {
+                        Book book = new Book(bookId, isbn, title, author, description, oldBook.getOwner(), BOOK_STATUS.AVAILABLE);
+                        // addImage will also replace if file with imageName already exist
+                        if (newUri != null) {
+                            // when the image is updated
```

Comment:
this will update the image even though it might not be changed, not sure if this is the best practice

Reply:
if it's not changed (they didn't click upload image btn) newUri is null. are you talking about when they uploaded the same image? that's IMO a scenario we can definitely ignore and set anyways.

Reply:
actually nvm, I think we only have newUri only if user uploads another one?

Reply:
exactly.